### PR TITLE
Add feature to unassign volunteer

### PR DESCRIPTION
**Pull request by ttzuche**
Partially handles #70 #71 #72 

- Added a new command `unassign` to VolunSync
- If volunteer is already unassigned or not involved in the event, a `CommandException` is thrown with the message:  
  `Error: Volunteer is not assigned to this event! `

### Changes:
- Added new class `UnassignCommand`:
  - Retrieve event and volunteer by index.
  - Check for valid indices for both volunteer and event.
  - Unassign volunteer and event only if they are assigned.

### Example Usage:
` unassign v/1 /e/2 `

### Behavior:
- When given invalid index:
    The possible errors are
  - ` Error: The volunteer index provided is invalid `
  - ` Error: The event index provided is invalid `
- On unsuccessful unassignment:  
  ` Error: Volunteer is not assigned to this event! `
- On successful unassignment:  
  ` Volunteer unassigned successfully! `

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/UnassignCommand.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+     */
+    public UnassignCommand(Index volunteerIndex, Index eventIndex) {
+        requireAllNonNull(volunteerIndex, eventIndex);
+        this.volunteerIndex = volunteerIndex;
+        this.eventIndex = eventIndex;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireAllNonNull(model);
```
Comment (Fluffykan) [inline src/main/java/seedu/address/logic/commands/UnassignCommand.java:52]: I think this cld just use requireNonNull(model), but no big issue even if its left as is.

File: src/main/java/seedu/address/logic/parser/UnassignCommandParser.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+import java.util.stream.Stream;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.UnassignCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+
+/**
+ * Parses input arguments and creates a new UnassignCommand object.
+ */
+public class UnassignCommandParser {
```
Comment (Fluffykan) [inline src/main/java/seedu/address/logic/parser/UnassignCommandParser.java:16]: This class should implement Parser&lt;UnassignCommand&gt;

**Review by Fluffykan — CHANGES_REQUESTED** [review]
LGTM, just need to make UnassignCommandParser implement Parser&lt;UnassignCommand&gt;

File: src/main/java/seedu/address/logic/commands/UnassignCommand.java [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+            throw new CommandException(MESSAGE_NOT_ASSIGNED);
+        }
+
+        if (!v.getEvents().contains(e.getName().toString())) {
+            throw new CommandException(MESSAGE_NOT_ASSIGNED);
+        }
+
+        e.removeVolunteer(v.getName().toString());
+
+        v.removeEvent(e.getName().toString());
```
Comment (keerthigkaarthik) [inline src/main/java/seedu/address/logic/commands/UnassignCommand.java:80]: Would it be better to do this inside address book instead? For the other commands, we don't do the actual logic inside execute itself.

File: src/main/java/seedu/address/model/event/Event.java [hunk 4]
```diff
@@ -96,6 +96,22 @@ public void addVolunteer(String newVolunteer) {
         volunteers.add(newVolunteer);
     }
 
+    /**
+     * Removes a volunteer from the event.
+     *
+     * @param removedVolunteer The name of the volunteer to remove.
+     * @return True if the volunteer was removed, false if they weren't found.
+     */
+    public boolean removeVolunteer(String removedVolunteer) {
```
Comment (keerthigkaarthik) [inline src/main/java/seedu/address/model/event/Event.java:105]: Should probably call this unassign instead. Same for all other 'remove'.

**Review by keerthigkaarthik — CHANGES_REQUESTED** [review]

**Review by Fluffykan — APPROVED** [review]
LGTM

**Review by keerthigkaarthik — APPROVED** [review]

**Pull request merged**